osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs

```diff
@@ -152,6 +140,11 @@ protected override ISkin GetSkin()
                     return null;
                 }
             }
+
+            ~BeatmapManagerWorkingBeatmap()
+            {
+                dereferenceAction?.Invoke(TrackStore);
+            }
```

Comment:
Historically this was required because we are not sure when a working beatmap was finished being used. I'm not sure if that has changed at all with this PR but I'd hope we have a more defined lifetime now.